src/tinuous/github.py

```diff
@@ -132,16 +134,41 @@ def get_event_id(self, run: WorkflowRun, event_type: EventType) -> str:
                     # have to fall back to performing an issue search to fill
                     # those in.  This should hopefully be used sparingly, as
                     # there's a 30 searches per hour rate limit.
-                    try:
-                        pr = str(
-                            self.client.search_issues(
-                                f"repo:{run.repository.full_name} is:pr {run.head_sha}",
-                                sort="created",
-                                order="asc",
-                            )[0].number
-                        )
-                    except IndexError:
-                        pr = "UNK"
+                    while True:
+                        try:
+                            pr = str(
+                                self.client.search_issues(
+                                    f"repo:{run.repository.full_name} is:pr"
+                                    f" {run.head_sha}",
+                                    sort="created",
+                                    order="asc",
+                                )[0].number
+                            )
+                            break
+                        except IndexError:
```

Comment:
note -- if `IndexError` is somehow raised within `search_issues` due to  a bug, we would react as we simply got no PR.  Ideally should be made more specific... I guess - could do later (sad to see that tests also do not hit `break` so seems to cover only the search failing to find any use case)